tests/test_authorization_code.py

```diff
@@ -537,6 +539,40 @@ def get_auth(self):
         query_dict = parse_qs(urlparse(response["Location"]).query)
         return query_dict["code"].pop()
 
+    def generate_pkce_codes(self, algorithm, length=43):
+        """
+        Helper method to generate pkce codes
+        """
+        code_verifier = get_random_string(length)
+        if algorithm == "S256":
+            code_challenge = base64.urlsafe_b64encode(
+                hashlib.sha256(code_verifier.encode()).digest()
+            ).decode().rstrip("=")
+        else:
+            code_challenge = code_verifier
+        return code_verifier, code_challenge
+
+    def get_pkce_auth(self, code_challenge, code_challenge_method):
+        """
+        Helper method to retrieve a valid authorization code using pkce
+        """
+        oauth2_settings.PKCE_REQUIRED = True
+        authcode_data = {
+            "client_id": self.application.client_id,
+            "state": "random_state_string",
+            "scope": "read write",
+            "redirect_uri": "http://example.org",
+            "response_type": "code",
+            "allow": True,
+            "code_challenge": code_challenge,
+            "code_challenge_method": code_challenge_method,
+        }
+
+        response = self.client.post(reverse("oauth2_provider:authorize"), data=authcode_data)
```

Comment:
I had a quick question about this. Shouldn't this be a GET request? I have been trying to get pkce to run against master but couldn't get  it to work. I would appreciate any pointers no how to get it working.